packages/block-library/src/search/index.php

```diff
@@ -442,7 +442,15 @@ function get_typography_styles_for_block_core_search( $attributes ) {
 
 	// Add typography styles.
 	if ( ! empty( $attributes['style']['typography']['fontSize'] ) ) {
-		$typography_styles[] = sprintf( 'font-size: %s;', esc_attr( $attributes['style']['typography']['fontSize'] ) );
+		$typography_styles[] = sprintf(
+			'font-size: %s;',
+			gutenberg_get_typography_font_size_value(
+				array(
+					'size' => esc_attr( $attributes['style']['typography']['fontSize'] ),
```

Comment:
Why does this use `esc_attr`? `$attributes['style']['typography']['fontSize']` isn't a HTML attribute.

Reply:
From memory, the search block needs to explicitly call `esc_attr` because it isn't using `get_block_wrapper_attributes`. When the other server-rendered blocks pass their styles through to that function, it winds up being escaped around this line in core: https://github.com/WordPress/wordpress-develop/blob/5ebe28966e5decbe1862c147bf98db0d8094038e/src/wp-includes/class-wp-block-supports.php#L213 — so the search block needs to do its own escaping before injecting.

Reply:
It looks like the Search block's `style` attributes are already being sanitised here using `safecss_filter_attr`:

https://github.com/WordPress/gutenberg/blob/6a2d304b3c66af697b4f535708c23e51f057115a/packages/block-library/src/search/index.php#L402-L405

That should mean all of the `esc_attr` calls in this function are unnecessary (albeit probably harmless).

At the very least let's change this so that `esc_attr` is called on the output of `gutenberg_get_typography_font_size_value`, not the input. It's not right at all to escape the input as `gutenberg_get_typography_font_size_value` wouldn't know what to do with an HTML escape sequence (e.g. `$lt; or `$quot;`) if it encountered one.

Reply:
> At the very least let's change this so that esc_attr is called on the output of gutenberg_get_typography_font_size_value, not the input.

Ah, yes, good point! Agreed.

> It looks like the Search block's style attributes are already being sanitised here using safecss_filter_attr:

`safecss_filter_attr` doesn't handle all circumstances for when a CSS string is used in an inline style, so we still need the `esc_attr` call.

Reply:
> `safecss_filter_attr` doesn't handle all circumstances for when a CSS string is used in an inline style, so we still need the `esc_attr` call.

I'd say move the `esc_attr` call up to `styles_for_block_core_search` then. Strangely that's how it works for `'label'` but none of the other keys.

Reply:
Good feedback, hadn't thought that `esc_attr` might return something the font size function can't process 😅 I'll fix  it!

Reply:
Done

Reply:
Thanks for fixing that up @tellthemachines! 🙇